### ISSUE TITLE
Unable to run Next.js React App

### ISSUE THREAD

**Issue by tgourley** [dotnet/aspire#5088]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I am modifying the existing sample Aspire project to run a React app to run a Next.js app.  The app will run independently utilizing Docker Compose, but I'm having an issue trying to run through Aspire.

### Steps To Reproduce

Docker Compose that works
```
version: "3.8"
services:
  frontend:
    container_name: frontend
    build:
      context: . #if your Dockerfile is not at the same level change the path here (./frontend)
      target: dev
    restart: always
    command: npm run dev
    environment:
      - NODE_ENV=development
    volumes:
      - .:/app
      - /app/node_modules
      - /app/.next
    ports:
      - 3000:3000
``` 

Aspire Setup
```
builder.AddNpmApp("web", "../frontend")
    .WithReference(minimalApi)
    .WithEnvironment("NODE_ENV", "development")
    .WithEnvironment("BROWSER", "none") // Disable opening browser on npm start
    .WithHttpEndpoint(env: "PORT")
    .WithExternalHttpEndpoints()
    .WithArgs("-v .:/app -v /app/node_modules -v /app/.next")
    .PublishAsDockerFile();
```

The image launches, with the following errors:
```
> next start
 
 ⚠ You are using a non-standard "NODE_ENV" value in your environment. This creates inconsistencies in the project and is strongly advised against. Read more: https://nextjs.org/docs/messages/non-standard-node-env
  ▲ Next.js 14.2.5
  - Local:        http://localhost:53811/
 
 ✓ Starting...
Error: Could not find a production build in the '.next' directory. Try building your app with 'next build' before starting the production server. https://nextjs.org/docs/messages/production-start-no-build-id
    at setupFsCheck (C:\Development\gitlab\one-page-finance\OnePageFinance.Web\node_modules\next\dist\server\lib\router-utils\filesystem.js:151:19)
    at async initialize (C:\Development\gitlab\one-page-finance\OnePageFinance.Web\node_modules\next\dist\server\lib\router-server.js:61:23)
    at async Server.<anonymous> (C:\Development\gitlab\one-page-finance\OnePageFinance.Web\node_modules\next\dist\server\lib\start-server.js:249:36)
```

### .NET Version info

.NET SDK:
 Version:           8.0.302
 Commit:            ef14e02af8
 Workload version:  8.0.300-manifests.5273bb1c
 MSBuild version:   17.10.4+10fbfbf2e

.NET SDKs installed:
  5.0.408 [C:\Program Files\dotnet\sdk]
  6.0.321 [C:\Program Files\dotnet\sdk]
  8.0.302 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.31 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.31 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.31 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

### Anything else?

The mapped volumes do exist and the Docker Compose runs without issue.
![image](https://github.com/user-attachments/assets/8ab42865-79c5-4555-bca7-f5145fbbf16b)

You will notice that I'm trying to pass in args to try and mount some volumes, this is just me trying to replicate what docker compose is doing since it is working.

**Comment by Alirexaa** [comment]
Could you change the build directory from `.next` to something like 'next-build' and try again?

**Comment by tgourley** [comment]
No luck with that change.

Here is the folder structure.  the `.next` folder is still created for some reason, but is empty.
![image](https://github.com/user-attachments/assets/b47a5c20-9c2e-4803-9c41-69ae9a6f6c8b)

Here is the `next.config.mjs`
```javascript
/** @type {import('next').NextConfig} */
const nextConfig = {
  distDir: "next-build",
};

export default nextConfig;
``` 

I didn't change anything in the Aspire app host.  I did need to change the dockerfile a little as it was copying over a couple of files from `.next`.  

Docker Compose still works with the change, but Aspire does not.

Updated error message
```
> next start
 
  ▲ Next.js 14.2.5
  - Local:        http://localhost:58132/
 
 ✓ Starting...
 ⚠ You are using a non-standard "NODE_ENV" value in your environment. This creates inconsistencies in the project and is strongly advised against. Read more: https://nextjs.org/docs/messages/non-standard-node-env
Error: Could not find a production build in the 'next-build' directory. Try building your app with 'next build' before starting the production server. https://nextjs.org/docs/messages/production-start-no-build-id
    at setupFsCheck (C:\Development\gitlab\one-page-finance\OnePageFinance.Web\node_modules\next\dist\server\lib\router-utils\filesystem.js:151:19)
    at async initialize (C:\Development\gitlab\one-page-finance\OnePageFinance.Web\node_modules\next\dist\server\lib\router-server.js:61:23)
    at async Server.<anonymous> (C:\Development\gitlab\one-page-finance\OnePageFinance.Web\node_modules\next\dist\server\lib\start-server.js:249:36)
```

**Comment by jack775544** [comment]
By the looks of it you are misunderstanding how the AddNpmApp function is working. It does not run the application in a docker container but instead runs it on your local computer. So the WithArgs is not needed since there is no docker.

In addition to that you are also running different commands for the Aspire and Docker versions. For Docker you are running `npm run dev` where for Aspire you are calling `.AddNpmApp("web", "../frontend")` which will default to running `npm run start`. You have to change the Aspire command to be `.AddNpmApp("web", "../frontend", "dev")` to make it run the correct script.

**Comment by tgourley** [comment]
@jack775544 you are correct.  I was misunderstanding how the `AddNpmApp` functions.  Using your change the NextJS React app ran as expected.  The only other issue I had is that the example `AspireWithJavaScript` example that Microsoft demonstrates React, relies on webpack to handle the proxy redirect between the web app and the .NET API.  NextJS doesn't use webpack, so I needed to do some work to switch the proxying over by using `http-proxy-middleware`.  Once I did that, I was able to have a working NextJS React app running with Aspire.  Thanks!
